#!/usr/bin/env python
PACKAGE = "lidar_perception"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("use_pca_box",            bool_t,   0, "Default: False",  False)
gen.add("use_tracking",           bool_t,   0, "Default: True",   True)

gen.add("voxel_grid_size",        double_t, 0, "Default: 0.2",    0.2,  0.0,  1.0)

gen.add("roi_max_x",              double_t, 0, "Default: 20",     70,   0,    100)
gen.add("roi_max_y",              double_t, 0, "Default: 10",     30,   0,    100)
gen.add("roi_max_z",              double_t, 0, "Default: 1",      1,    0,    3)

gen.add("roi_min_x",              double_t, 0, "Default: 0",    -30,  -100, 0)
gen.add("roi_min_y",              double_t, 0, "Default: -15",    -30,  -100, 0)
gen.add("roi_min_z",              double_t, 0, "Default: -2.40",   -2.5, -5,   0)

gen.add("ground_threshold",       double_t, 0, "Default: 0.3",    0.3,  0.0,  1.0)

gen.add("cluster_threshold",      double_t, 0, "Default: 1.005",    0.6,  0.0,  1.5)
gen.add("cluster_max_size",       int_t,    0, "Default: 1000",   5000, 0,    10000)
gen.add("cluster_min_size",       int_t,    0, "Default: 10",     10,   0,    100)

gen.add("displacement_threshold", double_t, 0, "Default: 1.0",    1.0,  0.0,  3.0)
gen.add("iou_threshold",          double_t, 0, "Default: 1.0",    1.0,  0.0,  1.0)

exit(gen.generate(PACKAGE, "obstacle_detector", "obstacle_detector"))